pkg/alertmanager/storage.go

```diff
@@ -0,0 +1,40 @@
+package alertmanager
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alerts"
+	"github.com/cortexproject/cortex/pkg/alertmanager/local"
+	"github.com/cortexproject/cortex/pkg/configs/client"
+)
+
+// AlertStoreConfig configures the alertmanager backend
+type AlertStoreConfig struct {
+	Type     string            `yaml:"type"`
+	ConfigDB client.Config     `yaml:"configdb"`
+	Local    local.StoreConfig `yaml:"local"`
+}
+
+// RegisterFlags registers flags.
+func (cfg *AlertStoreConfig) RegisterFlags(f *flag.FlagSet) {
+	cfg.Local.RegisterFlags(f)
+	cfg.ConfigDB.RegisterFlagsWithPrefix("alertmanager.", f)
+	f.StringVar(&cfg.Type, "alertmanager.storage.type", "configdb", "Method to use for backend alertmanager configs storage (configdb, file)")
```

Comment:
> Method to use for backend alertmanager configs storage

Maybe it's more clear if we describe it like `Type of backend to use to store alertmanager configs. Supported values are: "configdb", "local".` (pay attention "local", not "file")